modules/lib/src/main/resources/assets/js/app/ai/AI.ts

```diff
@@ -384,14 +400,28 @@ export class AI {
         return path.startsWith(AI.PAGE_PREFIX);
     }
 
+    private isPageComponentPath(path: string): boolean {
+        return this.isPagePath(path) && path.indexOf(AI.CONFIG_PREFIX) < 0;
+    }
+
     private isTopicPath(path: string): boolean {
         return path.startsWith(AI.TOPIC);
     }
 
     private getAiHelperByXData(path: string): AiHelper | undefined {
         const xData = this.getXData(path);
 
-        return xData?.xDataStepForm ? AiHelper.getAiHelpers().find((helper: AiHelper) => this.isHelperForXData(helper, xData)) : null;
+        return xData?.xDataStepForm ? AiHelper.getAiHelpersByGroup('xdata').find(
+            (helper: AiHelper) => this.isHelperForXData(helper, xData)) : null;
+    }
+
+    private getAiHelperByPage(path: string): AiHelper | undefined {
+        if (path.indexOf(AI.CONFIG_PREFIX) > -1) {
+            const dataPath = path.replace(AI.PAGE_PREFIX, '').split(`/${AI.CONFIG_PREFIX}`)[1];
+            return AiHelper.getAiHelpersByGroup('page').find((helper: AiHelper) => helper.getDataPath() === dataPath);
+        }
+
+        return null;
```

Comment:
Our TS config is not strict and won't actually detect if this function actually returns undefined. But, when we finally make it strict, `undefined` and `null` are not of the same type and result in error. Let it just work as "find" and return `undefined`, like you assumed in return value. 
```suggestion
    private getAiHelperByPage(path: string): AiHelper | undefined {
        if (path.indexOf(AI.CONFIG_PREFIX) > -1) {
            const dataPath = path.replace(AI.PAGE_PREFIX, '').split(`/${AI.CONFIG_PREFIX}`)[1];
            return AiHelper.getAiHelpersByGroup('page').find((helper: AiHelper) => helper.getDataPath() === dataPath);
        }

        return undefined;
```